### PR TITLE
Simplify useRenderCampaigns

### PR DESCRIPTION
**Pull request by timowestnosto**
Extracts useRenderCampaigns out of the main context and reduces the rerun conditions for the Session API side effects

### CODE REVIEW

File: src/components/NostoOrder.tsx [hunk 1]
```diff
@@ -38,7 +36,8 @@ export default function NostoOrder(props: {
         .load()
       renderCampaigns(data, api)
     },
-    [recommendationComponent, pageTypeUpdated]
+    [order],
```
Comment (timowestnosto) [inline src/components/NostoOrder.tsx:39]: order was missing

File: src/context.ts [hunk 2]
```diff
@@ -12,14 +16,7 @@ export interface NostoContextType {
...
   renderFunction?: AnyFunction
   responseMode: RenderMode
-  recommendationComponent?: React.ReactElement<{
-    nostoRecommendation: Recommendation
-  }>
-  useRenderCampaigns(type: string): {
-    renderCampaigns(data: unknown, api: NostoClient): void
-    pageTypeUpdated: boolean
-  }
-  pageType: string
```
Comment (timowestnosto) [inline src/context.ts:22]: not anymore used

File: src/context.ts [hunk 3]
```diff
@@ -12,14 +16,7 @@ export interface NostoContextType {
   currentVariation?: string
   renderFunction?: AnyFunction
   responseMode: RenderMode
-  recommendationComponent?: React.ReactElement<{
-    nostoRecommendation: Recommendation
-  }>
-  useRenderCampaigns(type: string): {
```
Comment (timowestnosto) [inline src/context.ts:18]: moved to standalone hook that uses context internally

File: src/hooks/useRenderCampaigns.tsx [hunk 4]
```diff
@@ -16,8 +16,10 @@ function RecommendationComponentWrapper(props: {
   })
 }
 
-function injectCampaigns(data: ActionResponse, api: NostoClient) {
-  api.placements.injectCampaigns(data.recommendations)
+function injectCampaigns(data: ActionResponse) {
+  window.nostojs(api => {
```
Comment (manikandan-ravikumar) [inline src/hooks/useRenderCampaigns.tsx:20]: Maybe include it under a check whether window.nostojs is present and log a warning that nostojs is not initialized?

**Review by TomJas-nosto — APPROVED** [review]

File: spec/navigation.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    <Link to="/collections/hoodies">Hoodies</Link>
+    <Link to="/">Home</Link>
+  </>
+}
+
+function Main() {
+  return <NostoProvider
+    account="shopify-11368366139"
+    recommendationComponent={<RecommendationComponent />}> 
+    <BrowserRouter>
```
Comment (manikandan-ravikumar) [inline spec/navigation.spec.tsx:48]: We need to keep in mind this only checks for navigation using the browser's history.

Comment (timowestnosto) [reply spec/navigation.spec.tsx:48]: can you elaborate?

**Review by manikandan-ravikumar — APPROVED** [review]

**Pull request merged**